### ISSUE TITLE
Observer does not seem to be working

### ISSUE THREAD

**Issue by sdevo619** [ArtSabintsev/Zephyr#31]
I am fairly familiar with using observers, but the function you have listed in the documentation does not seem to be working. I need to be able to immediately update the ui after I change a single string value in NSUserDefaults. I would prefer not sync all user defaults, but have been having a hell of time getting this to work consistently A detailed example would be spectacular if you may have one. I have gotten this to work, but I am sure there is a better way. I found that creating an observer for NSUbiquitousKeyValueStore.didChangeExternallyNotification seems to work as a workaround, but based on the data in the public override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?), I would assume it would be much easier/better to use this, as I hate adding extra observers if I can avoid it, so I don't have to worry about cleanup later on. Im sure you can appreciate that as much as myself. In case you are wondering I am testing on two actual devices, not the simulator. Also, as a separate question, on app startup when I add Zephyr.addKeysToBeMonitored(keys: ["somekey"]), do you still need to use Zephyr.sync(keys: ["somekey"]), or can I just begin to write to that specific UserDefault like normal, and Zephyr will take care of syncing it? These are the type of questions a detailed example will help illustrate better. I know it may seem very straightforward to you, but with no examples it is really hard to interpret how you intended this to be used. Thank you and keep up the great work. I would love to keep using this framework as it seems dead simple once I have some more insight into how you intend it to be used.

**Comment by ArtSabintsev** [comment]
Hi,

Sorry for the late response - lots going on. Would you mind reformatting your question and adding some syntax blocks. It's a huge block of text at the moment and a bit hard to follow as you added questions on questions in there.

**Comment by sdevo619** [comment]
Sorry about that. I cannot get the observer function to work that you have listed in your documentation. If you can just send me an example of how you intend that to be used to observe a string variable, that is all I need. Thank you very much for the response and for building Zephyr!!!

**Comment by ArtSabintsev** [comment]
I’m still confused. When you update UserDefaults the key should sync to iCloud. When something is changed on a different device and stored in iCloud, then and only then, will the change appear on your current device (e.g., the user defaults value will change) and the UI should update.

Make sure the key your interested in is set as a monitored key.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ArtSabintsev** [comment]
Closing this issue as I'm not sure how I'm going to be able to assist at all without a sample project and some more guidance.

Feel free to reopen if you need further assistance.
